### PR TITLE
CSAM: Adds routes types and rework Case components routing

### PR DESCRIPTION
**Pull request by GPaoloni**
This PR is part of "splitting https://github.com/techmatters/flex-plugins/pull/543 in smaller chunks".
Primary reviewer: @murilovmachado
Note: is easier to review commit-per-commit.

This PR 
- Isolates the routes where a `Case` can be present.
- Refines the routes being passed to the `Case` component.

### CODE REVIEW

File: plugin-hrm-form/src/states/routing/types.ts [hunk 1]
```diff
@@ -44,6 +44,16 @@ export type AppRoutes =
...
 
+export type CSAMReportRoute = {
+  route: 'csam-report';
+  subroute: 'form' | 'loading' | 'status';
+};
+
+type OtherRoutes = CSAMReportRoute;
+
+// The different routes we have in our app
+export type AppRoutes = AppRoutesWithCase | OtherRoutes;
```
Comment (murilovmachado) [inline plugin-hrm-form/src/states/routing/types.ts:55]: Great! Later on we should split the other route types, too.

**Review by murilovmachado — APPROVED** [review]
Looks good!

File: plugin-hrm-form/src/components/case/Case.tsx [hunk 2]
```diff
@@ -529,7 +539,6 @@ const Case: React.FC<Props> = props => {
                 Icon={CancelIcon}
                 text={<Template code="BottomBar-CancelNewCaseAndClose" />}
                 onClick={handleCancelNewCaseAndClose}
-                data-fs-id="Case-SaveAndEnd-Button"
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/case/Case.tsx:532]: Not my fullstory IDs!! Do you just need to merge in from master?

**Review by stephenhand — APPROVED** [review]
A bit lost in this one I have to admit, I'll be able to review PRs in this area better when I understand the internal routing / redux stuff in the plugins a little more.

File: plugin-hrm-form/src/components/case/Case.tsx (hunk 2 above)
Comment (GPaoloni) [reply plugin-hrm-form/src/components/case/Case.tsx:535]: Yes this was solved after merging master into the branch. Thanks for pointing!

**Pull request merged**